Yoast/wordpress-seo
# Use meaningful strings for better translations

## Summary

This PR can be summarized in the following changelog entry:

* Improved Content Analysis and Publish box strings for better translations.

As mentioned in the related issue, some strings need to give translators the ability to translate them using the proper genre, singular/plural forms. To achieve this, we should always provide context or full, meaningful, strings. This PR tries to avoid strings that are composed by multiple variables or single words without context and provides full string with an understandable context.
Any feedback and changes welcome!

Question: is the bold text in the Publish box really necessary? Code could be much cleaner without that bold.

Note: also removes a couple ESLint warnings to stay under the 106 max warnings limit.

## Test instructions

- Build the JS.
- Edit several posts with different readability and SEO scores
- check in the Publish box the readability and SEO scores strings are always correct
- check the metabox content analysis headings are now changed in `Analysis results` and `Good results`. I haven't changed the strings for problems, improvements and considerations. DOn't forget to check the metabox in the taxonomies (the media metabox should be the same as posts, right?).

Fixes #8532
